packages/dev/s2-docs/src/TitleResizer.tsx

```diff
@@ -0,0 +1,75 @@
+'use client';
+
+import {useCallback, useRef} from 'react';
+import {useLayoutEffect} from '@react-aria/utils';
+
+// TODO: this causes a flash when the headings change size, maybe we just move this code
+// to client and append suppressHydrationWarning to the h1s
+export function TitleResizer() {
```

Comment:
Want to use https://una.im/5-css-functions/#fluid-type if it's supported in the browser?
it's already in chrome https://github.com/Fyrd/caniuse/issues/7368